README.md

```diff
@@ -0,0 +1,112 @@
+
+sttp is a family of Scala HTTP-related projects, and currently includes:
+
+* [sttp client](https://github.com/softwaremill/sttp): The Scala HTTP client you always wanted!
+* [sttp tapir](https://github.com/softwaremill/tapir): Typed API descRiptions
+* sttp openai: this project. Scala client wrapper for OpenAi API. Use the power of ChatGPT inside your code!
+
+## Quickstart with sbt
+
+Add the following dependency:
+
+[//]: # (```scala)
+[//]: # ("com.softwaremill.sttp.openai" %% "core" % "?.?.?")
+[//]: # (```)
+
+sttp openai is available for Scala 2.12 and Scala 3.2.2
+
+## Project content
+
+Available endpoints with modeled response classes include:
+* Models:
+  * Retrieves all the available models
+  * Retrieves model for given ID
+* Completions
+  * Creates a completion for provided model and prompt
+* Chat
+  * Creates a completion for the provided prompt and parameters
+* Edits
+  * Creates a new edit for the provided input, instruction, and parameters
+* Images
+  * Creates an image given a prompt (Returns an url with an image)
+  * Creates an edited or extended image given an original image and a prompt (Returns an url with an image)
+  * Creates a variation of a given image (Returns an url with an image)
+* Embeddings
+  * Creates an embedding vector representing the input text
+* Audio
+  * Transcribes audio into the input language
+  * Translates audio into English
+* Files
+  * Returns a list of files that belong to the user's organization
+  * Upload a file that contains document(s) to be used across various endpoints/features
+  * Delete a file for given ID
+  * Returns information about a specific file providing file's ID
+  * Returns the contents of the specified file providing file's ID
+* Fine-Tunes
+  * Creates a job that fine-tunes a specified model from a given dataset
+  * Retrieves all your organization's fine-tuning jobs
+  * Retrieves info about the fine-tune job providing fine-tune's ID
+  * Immediately cancel a fine-tune job providing fine-tune's ID
+  * Retrieves fine-grained status updates for a fine-tune job providing fine-tune's ID
+  * Delete a fine-tuned model providing fine-tune's ID
+* Moderation
+  * Classifies if text violates OpenAI's Content Policy
+
+## Example
+
+### To use ChatGPT
+
+```scala mdoc:compile-only 
+import sttp.client4._
+import sttp.openai.requests.completions.chat.ChatRequestResponseData.ChatResponse
+import sttp.openai.requests.completions.chat.ChatRequestBody.ChatBody
+import sttp.openai.requests.completions.chat.Message
+
+// Create an instance of OpenAi providing your API secret-key
+
+val openAi: OpenAi = new OpenAi("your-secret-key")
+
+// Create body of Chat Completions Request
+
+val bodyMessages: Seq[Message] = Seq(
+  Message(
+    role = "user",
+    content = "Hello!"
+  )
+)
+
+val chatRequestBody: ChatBody = ChatBody(
+  model = "gpt-3.5-turbo",
```

Comment:
shouldn't this be an enum constant?

Reply:
It can be, but then we have to change 
```
  case class ChatBody(
      model: String, ...)
```
To ADT smth like:
```
  case class ChatBody(
      model: ChatModel, ...)
```
According to docs there are different models to different functionality. 
https://platform.openai.com/docs/models/model-endpoint-compatibility

So we can either release it with String and make another release with missing models ADT, wdyt?

Reply:
hm so the same `ChatBody` is used for different endpoints, and each endpoint has a limited subset of models?

Reply:
ChatBody is used only for createChatCompletion method.
Other methos that uses other type of models should have their own ADT. For example in Edits endpoint we create CompletionsBody
```
  case class CompletionsBody(
      model: String ...
```
uses string for now, but we can use ADT, that we should  explicity name as `EditsModel` because API accepts only one of models for Edits, which are text-davinci-edit-001 and code-davinci-edit-001 and those models are not shared with other endpoints/functionality

Reply:
I mean, without a doubt we should be using an ADT here as we are using in other parts of the project already. The question is, do we want to realease it first and update it in next release or postpone the relase and "fix" it first

Reply:
Sure, let's create an issue an fix it later.